### DISCUSSION TITLE
TP 5 Ejercicio 1 - prototipos de las funciones

### DISCUSSION THREAD

**Discussion by TeoInMaghine** [INGCOM-UNRN-P1/cursada-2023#169] [Preguntas y respuestas] [answered]
Buenas. En el ejercicio 1 de matrices se nos da prototipos para las funciones para implementar:
```
int producto_escalar(int filas, int columnas, int matriz[][], int escalar);
```
y
```
int suma_matriz(int filas_1, int columnas_1, int matriz_1[][],
								int filas_2, int columnas_2, int matriz_2[][],
								int filas_r, int columnas_r, int resultado[][]);
```

Pero no deja compilar si declaro las matrices así, me pone:
```
ejercicio1.c:25:51: error: array type has incomplete element type 'int[]'
   25 | int producto_escalar(int filas, int columnas, int matriz[][], int escalar)
      |                                                   ^~~~~~
ejercicio1.c:25:51: note: declaration of 'matriz' as multidimensional array must have bounds for all dimensions except the first
```
---
Entonces, mis dudas serían:
1. ¿Por qué está así la consigna?
2. ¿Qué acercamiento se supone que tome? Se me ocurre definir con macros ya el tamaño de las matrices, pero en ese caso para qué pasar la cantidad de filas y columnas como argumento; ó usar un tamaño máximo pre-definido, y pasar los tamaños más pequeños por argumento para limitar los for.

**Comment by martinvilu** [comment, marked as answer]
Lo referido al prototipo de las funciones, lo vimos ayer en clase, es necesario indicar el ancho de la matriz y para esto, se puede usar el argumento o la constante.

Lo segundo, es lo mismo que en los ejercicios anteriores, un tamaño fijo en compilación.

**Reply by TeoInMaghine** [reply]
Okayyy.

Mando acá el video de la clase con el tiempo que se habla sobre esto para el que quiera verlo en más detalle también -> https://youtu.be/WTzPsWQU_qc?si=2yaD1-YmsqWn16Sh&t=2939.

**Reply by martinvilu** [reply]
Si tienen estas marcas de tiempo, las puedo agregar a la descripción del video para que aparezcan en la navegación del video mismo.

**Reply by martinvilu** [reply]
Avisá si falta explicar algo, si hay algo que no se vea o algo que no se termine de entender.

**Comment by martinvilu** [comment]
Una de las formas indicadas en clase
```diff
-int producto_escalar(int filas, int columnas, int matriz[][], int escalar);
+int producto_escalar(int filas, int columnas, int matriz[][columnas], int escalar);
```
En donde `columnas` del argumento `matriz` es el argumento mismo. (que, de hecho, no funciona si el argumento `matriz` está antes)

La otra es usar la `CONSTANTE` definida.
